core/devdoc/message_queue_requirements.md

```diff
@@ -0,0 +1,128 @@
+MESSAGE QUEUE REQUIREMENTS
+==========================
+
+Overview
+--------
+
+The message queue is a very simple queue intended to manage messages. The message queue is typed with MESSAGE_HANDLE because the destruction of the queue requires the destruction of the messages inside the queue.  **Unless the queue is destroyed, the user of this queue is expected to clone before pushing onto the queue, and is expected to destroy the message after popping the message off teh queue.**
```

Comment:
I admit, I wanted a more easily mockable queue so that I didn't need to worry about the out of process module unit tests being even longer and more complicated than they already are.  This did help, but this unit is pretty basic.  I'm gonna keep it for now, with an eye to remove it if I need to make changes to how the queuing works in the out of process module.